include/boost/geometry/views/detail/normalized_view.hpp

```diff
@@ -36,72 +36,99 @@ namespace boost { namespace geometry {
+    using iterator = typename boost::range_iterator<view const>::type;
+    using const_iterator = typename boost::range_iterator<view const>::type;
 
-    typedef typename detail::range_type<Geometry>::type range_type;
+    inline const_iterator begin() const { return boost::begin(m_view); }
+    inline const_iterator end() const { return boost::end(m_view); }
 
-    typedef std::conditional_t
-        <
-            is_const,
-            range_type const,
-            range_type
-        > range;
-
-    typedef typename
-        reversible_view
-            <
-                range,
-                order_as_direction
-                    <
-                        geometry::point_order<Geometry>::value
-                    >::value
-            >::type reversible_type;
-
-    typedef std::conditional_t
-        <
-            is_const,
-            reversible_type const,
-            reversible_type
-        > reversible;
-
-    typedef typename
-        closeable_view
-            <
-                reversible,
-                geometry::closure<Geometry>::value
-            >::type closeable_type;
-
-    typedef std::conditional_t
+private:
+    view m_view;
+};
+
+
+template
+<
+    typename Geometry,
+    closure_selector Closure = geometry::closure<Geometry>::value,
+    iterate_direction Direction = geometry::order_as_direction
         <
-            is_const,
-            closeable_type const,
-            closeable_type
-        > closeable;
+            geometry::point_order<Geometry>::value
+        >::value
+>
+struct close_reverse_view
+{
+    // These utilities are inconsistent since Closure defines the property of
+    // the Geometry while Direction defines what to do with the Geometry.
+    using close_view = detail::close_view<Geometry const, Closure>;
+    using view = detail::reverse_view<close_view const, Direction>;
     
-    explicit inline normalized_view(range & r)
-        : m_reversible(r)
-        , m_closeable(m_reversible)
+    explicit inline close_reverse_view(Geometry const& g)
+        : m_view(close_view(g))
     {}
 
-    typedef typename boost::range_iterator<closeable>::type iterator;
-    typedef typename boost::range_const_iterator<closeable>::type const_iterator;
+    using iterator = typename boost::range_iterator<view const>::type;
+    using const_iterator = typename boost::range_iterator<view const>::type;
 
-    inline const_iterator begin() const { return boost::begin(m_closeable); }
-    inline const_iterator end() const { return boost::end(m_closeable); }
-
-    inline iterator begin() { return boost::begin(m_closeable); }
-    inline iterator end() { return boost::end(m_closeable); }
+    inline const_iterator begin() const { return boost::begin(m_view); }
+    inline const_iterator end() const { return boost::end(m_view); }
 
 private:
-    reversible_type m_reversible;
-    closeable_type m_closeable;
+    view m_view;
 };
 
+
+// Other possible names:
+//   clockwise_closed_view
```

Comment:
Why do we need it actually? Because it looks identical to the `reverse_close_view`

And indeed, a `clockwise_closed_view` is clockwise and closed, and that looks intuitive to me.

Reply:
I described the reasoning at the bottom of the PR description. There are both views because in some cases Direction is used and in other cases Order is used. With Order semantically corresponding to Closure. In practice some algorithms pass `Reverse` template argument which is then translated to `Direction` while in other algorithms the `Order` is taken from Geometry and this is translated to `Direction`. So to avoid translations in many algorithms I added these redundant views doing the translation internally.

Personally I think we need only one view. The one that first closes the geometry and then reverses it. There are two problems with this:
- in some algorithms the different order of operations is performed so we have to check if the algorithm fails or not, e.g. `buffer()`
- in many algorithms `Reverse` template argument is passed explicitly instead being taken from Geometry in place where it's needed
After both of these are changed and tested we could have only one view. Furthermore, Order->Direction translation will not be needed anymore.

Reply:
I would welcome one view. We most probably might remove the `Reverse` template argument in those cases.
So I used another order of operations in buffer, I was not aware of that. But it should result into the same source geometry, and therefore in the same buffer...

Reply:
The different starting point is generated only for open, ccw rings and AFAIK we mainly test with closed cw rings.